examples/gocean/eg6/Makefile

```diff
@@ -61,34 +66,36 @@ $(DRIVER_INIT):	$(KERNELS) $(DRIVER_INIT).o main.o \
 	$$(nf-config --flibs)
 
 # The driver does not yet compile because of #644
-$(DRIVER_UPDATE):	$(KERNELS) $(DRIVER_UPDATE).o main.o \
+$(DRIVER_UPDATE): $(KERNELS) $(DRIVER_UPDATE).o main.o \
 				$(EXTRACT_DIR)/lib_kernel_data_netcdf.a
 	$(F90) $(KERNELS) $(DRIVER_UPDATE).o main.o \
 	-o $(DRIVER_UPDATE)            \
 	-L$(EXTRACT_DIR) -l_kernel_data_netcdf \
 	$$(nf-config --flibs)
 
-
-$(EXEC):	$(KERNELS) alg.o psy.o $(EXTRACT_DIR)/lib_kernel_data_netcdf.a
-	$(F90)  $(KERNELS) alg.o psy.o -o $(EXEC)                   \
+$(EXEC): $(KERNELS) $(EXTRACT_DIR)/lib_kernel_data_netcdf.a alg.o psy.o
+	$(F90)  $(KERNELS) alg.o psy.o -o $(EXEC) \
 	-L$(EXTRACT_DIR) -l_kernel_data_netcdf \
 	-L $(DL_DIR) -l_fd                     \
 	$$(nf-config --flibs)
-	
 
-alg.f90 psy.f90: test.x90 extract_transform.py
-	psyclone -nodm -l -api "gocean1.0" -s ./extract_transform.py\
-			 --config ../../../config/psyclone.cfg \
+transform: test.x90 extract_transform.py
+	${PSYCLONE} -nodm -l -api "gocean1.0" -s ./extract_transform.py\
 		     -opsy psy.f90 -oalg alg.f90 test.x90
 
+psy.f90 alg.f90: transform
+
 alg.o:	psy.o 
 
 psy.o:	$(KERNELS)
 
 %.o: %.f90
 	$(F90) $(F90FLAGS) -I $(DL_DIR) -I $(EXTRACT_DIR) -c $<
 
-clean:
-	rm -f *.o *.mod alg.f90 psy.f90 $(EXEC)
-	rm -f $(DRIVER_INIT) $(DRIVER_INIT).f90 \
-		  $(DRIVER_UPDATE) $(DRIVER_UPDATE).f90
+# This target requires that the netcdf (Fortran) development package be
+# installed
+$(EXTRACT_DIR)/lib_kernel_data_netcdf.a:
+	make -C $(EXTRACT_DIR)
+
+allclean: clean
```

Comment:
Mmm, that's a tricky one. I wanted to make sure that a user can clean and then build all of the examples from a single location. One potential solution would be to alter the extraction library so it can be built 'outside' its source tree?

Reply:
See comment below.